### DISCUSSION TITLE
hook C++ 成员函数时会发生成员变量空指针解引用问题。

### DISCUSSION THREAD

**Discussion by Zheng-JiaCheng** [bytedance/android-inline-hook#41] [Q&A]
hook C++ 成员函数时会发生成员变量空指针解引用问题。

**Comment by caikelun** [comment]
根据C++成员函数的调用约定，运行时第一个参数是this指针，所以你需要在proxy函数参数中自己加上 `void *thiz`，调用原函数时也要传递过去。C++语法和运行时的区别需要使用者自己处理的。
